src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_STEP = 1.5;
+    private int size;
+    private T[] data;
+
+    public ArrayList(int capacity) {
+        if (capacity < 0) {
+            throw new IllegalArgumentException("Illegal Capacity: " + capacity);
+        }
+        this.data = (T[]) new Object[capacity];
+    }
+
+    public ArrayList() {
+        this.data = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void grow() {
+        T[] arrayCopy = (T[]) new Object[(int) (data.length * INCREASE_STEP)];
+        System.arraycopy(data,0,arrayCopy,0,data.length);
+        data = arrayCopy;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Passed index is invalid");
+        }
+    }
+
```

Comment:
grow() should be private 
private methods should go after public

Reply:
Thanks, my bad